.ci/install.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -ex
+
+if [[ "$(uname -s)" == 'Darwin' ]]; then
```

Comment:
I don't use a Mac.  Is it normal to have to special case everything for it?

Reply:
Yes. This is because of the following reason:

- OSX can *only* be built natively (no Docker)
- Using one `install` script is usually more comfortable

So we run the same script on multiple setups (Linux/Docker and OSX) with the effect of it is setting up a build toolchain (compiler+cmake+conan in this case).